evm/src/main/java/com/horizen/evm/interop/EvmParams.java

```diff
@@ -9,11 +9,13 @@ public class EvmParams extends HandleParams {
     public Address to;
     public BigInteger value;
     public byte[] input;
-    public BigInteger gasLimit; // uint64
+    public BigInteger gas; // uint64
```

Comment:
The difference between `gas` and `gasLimit` is subtle: The "gas limit" is provided from the outside, either via a transaction or during an RPC call. After the "buy gas" step during a state transition we have a certain amount of "gas" available that we consume until it's used up (out of gas) or we're done processing. At this stage in the process we are behind the "buy gas" step, thus using `gas` instead of `gasLimit` here is more accurate. Also this is not the original gas limit anymore as intrinsic gas was already paid.

This also reduces confusion with the `gasLimit` that is still existing on the context, which is the block gas limit, not the TX gas limit.

Reply:
thanks for the explanation.
I propose to give more context to the names:
* `params.gas` -> `params.availableGas` (or even better name)
* `gas` as a result of apply -> `gasLeft`
In this way `gasUsed = availableGas - gasLeft` seems more clear than `gasUsed = params.Gas - gas`

Reply:
still actual